app/code/Magento/SalesRule/Model/Validator.php

```diff
@@ -409,6 +403,33 @@ public function initTotals($items, Address $address)
         return $this;
     }
 
+    /**
+     * Determine if quote item is valid for a given sales rule
+     *
+     * @param AbstractItem $item
+     * @param Rule $rule
+     * @return bool
+     */
+    protected function isValidItemForRule(
```

Comment:
Could you make it for private and defined in one row?

Reply:
Will do :+1: